spot-webdriver/user/spot-user.js

```diff
@@ -105,13 +105,38 @@ class SpotUser {
      *
      * @returns {void}
      */
-    stopP2P() {
-        // to be implemented
+    stopP2PConnection() {
         this.driver.execute(rcsName => {
-            window.spot[rcsName]._p2pSignaling.stop();
+            try {
+                window.spot[rcsName]._p2pSignaling.stop();
+            } catch (e) {
+                // Error means p2p is already destroyed.
+            }
         }, this._internalRemoteControlServiceName);
     }
 
+    /**
+     * Stops messages from being sent to or received from the signaling service.
+     *
+     * @returns {void}
+     */
+    stopSignalingConnection() {
+        this.driver.setNetworkConditions(_OFFLINE_NETWORK_CONDITIONS);
```

Comment:
I would rather have the network offline call in the `setNetworkOffline` method, because that's how the method is called. This one could be renamed to 'waitForSignalingConnectionStopped`/`disconnected`.

Reply:
The test needs to stop signaling while keeping p2p alive. Now stop network kills both signaling and p2p. If stop network should continue to kill both, then another method will be needed to kill only signaling, which is how this stopSignalingConnection method came to be. How do you think signaling by itself should be killed?

Reply:
Maybe the changes you want are there now. `setNetworkOffline` is back to only changing network conditions. I made a helper for the repeated killing of xmpp and p2p.